apps/callscreen/js/handled_call.js

```diff
@@ -382,30 +362,37 @@ HandledCall.prototype.formatPhoneNumber =
 };
 
 HandledCall.prototype.replacePhoneNumber =
-  function hc_replacePhoneNumber(phoneNumber, ellipsisSide) {
-    this.numberNode.textContent = phoneNumber;
-    this.formatPhoneNumber(ellipsisSide);
+  function hc_replacePhoneNumber(number, ellipsisSide) {
+    if (typeof(number) === 'string') {
+      navigator.mozL10n.setAttributes(this.numberNode, number);
+    } else if (number.id) {
+      navigator.mozL10n.setAttributes(this.numberNode, number.id,
+                                      number.args);
+      this.numberNode.style.fontSize = '';
+    } else if (number.raw) {
+      this.numberNode.removeAttribute('data-l10n-id');
+      this.numberNode.textContent = number.raw;
+      this.formatPhoneNumber(ellipsisSide);
+    }
+
+    this._cachedInfo = number;
 };
 
 HandledCall.prototype.restorePhoneNumber =
   function hc_restorePhoneNumber() {
-    this.numberNode.textContent = this._cachedInfo;
-    this.formatPhoneNumber('end');
+    this._cachedInfo && this.replacePhoneNumber(this._cachedInfo, 'end');
 };
 
 HandledCall.prototype.updateDirection = function hc_updateDirection() {
-  var self = this;
   var classList = this.node.classList;
   if (this._initialState === 'incoming') {
     classList.add('incoming');
-    LazyL10n.get(function localized(_) {
-      self.node.setAttribute('aria-label', _('incoming'));
-    });
+    // XXX: This should be replaced with mozL10n.setAttribute whenever possible
+    this.node.setAttribute('aria-label', 'incoming');
```

Comment:
this will not do any l10n, right? Why not move the strings to `incoming.ariaLabel` and `outgoing.ariaLabel` and set `data-l10n-id` to `incoming`|`outgoing` here? You don't need to change the ID because our tools will catch this change.